lib/developer_portal/app/controllers/developer_portal/admin/account/personal_details_controller.rb

```diff
@@ -61,4 +63,17 @@ def redirect_path
     end
   end
 
+  def user_params
+    params.require(:user)
+  end
+
+  def verify_current_password
+    return unless current_user.using_password?
+
+    unless current_user.authenticated?(user_params[:current_password])
+      flash.now[:error] = 'Current password is incorrect.'
+      assign_user_drop
+      render :action => 'show'
```

Comment:
This looks like duplicated logic. It should be part of `update` somehow. Maybe add a validator for password if that's even possible or extend `update!` and return an error so that it can be catched with https://github.com/3scale/porta/blob/8612ff9ee902063a0e4c45dc49a4ffa2133e161b/lib/developer_portal/app/controllers/developer_portal/admin/account/personal_details_controller.rb#L28-L31